File: cpp/0763划分字母区间.cpp
#include<vector>
#include<string>

using namespace std;

vector<int> partitionLabels(string s)
{
    vector<int> res;
    int count = 0;
    int memo[26] = {0};
    int furthest = -1;
    int idx;
    for (int i = 0; i < s.size(); i++)
    {
        idx = s[i] - 'a';
        if (i > memo[idx])
            memo[idx] = i;
    }

    for (int i = 0; i < s.size(); i++)
    {
        count++;
        idx = s[i] - 'a';
        furthest = max(furthest, memo[idx]);
        if (i == furthest)
        {
            res.push_back(count);
            count = 0;
        }
    }
    return res;
}

File: python/0763划分字母区间.py

def partitionLabels(s: str) -> list:

    res=[]
    memo=[-1]*26 # 统计字母在s中最后出现的位置
    count=0 # 统计当前分片的元素个数
    furthest=-1 #当前分片最远重复元素位置

    for i in range(len(s)):
        idx=ord(s[i])-ord('a')
        if(i>memo[idx]):
            memo[idx]=i
    
    for i in range(len(s)):
        idx=ord(s[i])-ord('a')
        furthest=max(furthest,memo[idx])
        count+=1

        if(i==furthest):
            res.append(count)
            count=0

    return res

print(partitionLabels("ababcbacadefegdehijhklij"))